src/app/src/config.cpp

```diff
@@ -201,14 +200,14 @@ Config::Config(std::filesystem::path cfg, cxxopts::ParseResult userargs) {
     trigDev = toml::find_or<std::string>(config, "device", "nidaqmx", "device_2", std::string("DevOR"));
     if (userargs.count("trig_dev")) { trigDev = userargs["trig_dev"].as<std::string>(); }
 
-    maxVoltage = toml::find_or<double>(config, "device", "nidaqmx", "max_voltage", 1.4);
+    maxVoltage = toml::find_or<double>(machineVars, "device", "nidaqmx", "max_voltage", toml::find_or<double>(config, "device", "nidaqmx", "max_voltage", 1.4));
     if (userargs.count("max_voltage")) { maxVoltage = userargs["max_voltage"].as<double>(); }
 
-    shutterDelayMs = toml::find_or<uint16_t>(config, "device", "nidaqmx", "shutter_delay_ms", 1000);
+    shutterDelayMs = toml::find_or<uint16_t>(machineVars, "device", "nidaqmx", "max_voltage", toml::find_or<uint16_t>(config, "device", "nidaqmx", "shutter_delay_ms", 1000));
     if (userargs.count("shutter_delay_ms")) { shutterDelayMs = userargs["shutter_delay_ms"].as<uint16_t>(); }
 
     //device.tango
-    stageComPort = toml::find_or<std::string>(config, "device", "tango", "com", std::string("COM3"));
+    stageComPort = toml::find_or<std::string>(machineVars, "device", "tango", "com", toml::find_or<std::string>(config, "device", "tango", "com", std::string("COM3")));
```

Comment:
I think it probably doesn't make sense to have a fallback value for this, if the device com port isn't set in either file there's no way the software is going to have a good guess at what it is, it should probably pop up an error and tell the user there's no value.

Reply:
Okay I'll make a popup